app/redux/selectors/SetsSelectors.js

```diff
@@ -259,4 +261,21 @@ export const hasChangesToSync = (state) => {
     return (root.setIDsToUpload.length > 0 || root.setIDsBeingUploaded.length > 0);
 };
 
+// collapsed metrics
+
+export const getBestAvgVelocityEver = (state, set) => {
+    let historySets = getHistorySetsChronological(state);
+
+    // find all instances of this exercise with weight and reps
+    let matchedSets = historySets.filter(historySet => historySet.exercise === set.exercise && historySet.weight === set.weight &&  historySet.metric === set.metric && historySet.reps.length === set.reps.length);
+
+    let avgVs = matchedSets.map((matchedSet) => {
+        CollapsedMetrics.getAvgVelocities(matchedSet);
+    });
+
+    avgVs = avgVs.reduce((a, b) => a.concat(b), []);
+    
+    return Math.max(...avgVs);
```

Comment:
- [ ] fix this, it shouldn't work because reduce reduces to a single number and you'redoing a max on it